balancer/endpointsharding/endpointsharding_test.go

```diff
@@ -159,6 +162,13 @@ func (s) TestEndpointShardingBasic(t *testing.T) {
 	if err = roundrobin.CheckRoundRobinRPCs(ctx, client, []resolver.Address{{Addr: backend1.Address}, {Addr: backend2.Address}}); err != nil {
 		t.Fatalf("error in expected round robin: %v", err)
 	}
+
+	// Execute the resolver error path. pickfirst will ignore the resolver error
+	// as it has a good resolver state.
+	mr.ReportError(errors.New("test error"))
```

Comment:
This tests that the operation was a nop, which isn't really a good test of the functionality.  I.e. the child could not even get the call and the test would still pass.  I think it would be better to have a test that calls ResolverError before it's working and ensures that it results in pick errors / TF state.

Reply:
The children of enpointsharding wouldn't get created if the resolver doesn't produce a good list of endpoints first. 
I can use a stub child balancer which modifies pickfirst's behaviour of ignoring resolver errors when it has a working state.

Reply:
I stopped the backends, reported a resolver error and verified the RPC error message.